analyzers/src/SonarAnalyzer.CSharp/Rules/PureAttributeOnVoidMethod.cs

```diff
@@ -31,5 +32,19 @@ namespace SonarAnalyzer.Rules.CSharp
     public sealed class PureAttributeOnVoidMethod : PureAttributeOnVoidMethodBase<SyntaxKind>
     {
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            base.Initialize(context);
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    if (InvalidPureDataAttributeUsage(c.SemanticModel.GetDeclaredSymbol(c.Node)) is { } pureAttribute)
```

Comment:
could you first check the Attribute name at AST-level? To avoid querying symbols for all local functions (even those with no attributes or with attributes which are not `[Pure]` or `[PureAttribute]`)